BoxContentSDK/BoxContentSDK/Categories/UIApplication+ExtensionSafeCompilation.m

```diff
@@ -0,0 +1,44 @@
+//
+//  UIApplication+ExtensionSafeCompilation.m
+//
+//  Copyright © 2016 Box. All rights reserved.
+//
+
+#import "UIApplication+ExtensionSafeCompilation.h"
+
+@implementation UIApplication (ExtensionSafeCompilation)
+
++ (UIApplication *)box_sharedApplication
+{
+#ifdef TARGET_IS_EXTENSION
+    // If we are compiling from an extension, return nil here since UIApplication isn't extension-safe.
+    return nil;
+#else
+    // If we are compiling from a non-extension target, use the regular sharedApplication.
+    return [UIApplication sharedApplication];
+#endif
+}
+
+- (BOOL)box_canOpenURL:(NSURL *)url
```

Comment:
why do we need these? If you call box_sharedApplication you're set. Is there a reason we can't use box_SharedApplication in some cases?